components/post_view/post_list.jsx

```diff
@@ -347,43 +333,45 @@ export default class PostList extends React.PureComponent {
     }
 
     checkBottom = () => {
-        if (!this.refs.postlist) {
+        if (!this.postListRef.current) {
             return true;
         }
+        const postlistScrollHeight = this.postListRef.current.scrollHeight;
+        const postlistClientHeight = this.postListRef.current.clientHeight;
 
         // No scroll bar so we're at the bottom
-        if (this.refs.postlist.scrollHeight <= this.refs.postlist.clientHeight) {
+        if (postlistScrollHeight <= postlistClientHeight) {
             return true;
         }
 
-        return this.refs.postlist.clientHeight + this.refs.postlist.scrollTop >= this.refs.postlist.scrollHeight - CLOSE_TO_BOTTOM_SCROLL_MARGIN;
+        return postlistClientHeight + this.postListRef.current.scrollTop >= postlistScrollHeight - CLOSE_TO_BOTTOM_SCROLL_MARGIN;
     }
 
     handleWindowResize = () => {
         this.handleResize();
     }
 
     handleResize = (forceScrollToBottom) => {
-        const postList = this.refs.postlist;
-        const messageSeparator = this.refs.newMessageSeparator;
-        const doScrollToBottom = this.atBottom || forceScrollToBottom;
-
-        if (postList) {
-            if (doScrollToBottom) {
-                postList.scrollTop = postList.scrollHeight;
-            } else if (!this.hasScrolled && messageSeparator) {
-                const element = ReactDOM.findDOMNode(messageSeparator);
-                element.scrollIntoView();
-            }
+        if (this.resizeAnimationFrame) {
+            window.cancelAnimationFrame(this.resizeAnimationFrame);
+        }
 
-            this.previousScrollHeight = postList.scrollHeight;
-            this.previousScrollTop = postList.scrollTop;
-            this.previousClientHeight = postList.clientHeight;
+        this.resizeAnimationFrame = window.requestAnimationFrame(() => {
```

Comment:
This gets called quite a few times if there is not postmeta data and it will stop the scroll because of JS thread lock so adding animation frame request so as to let user continue his actions.

Reply:
About this yes, I think it is better to have this as i don't see any downsides here and it helps in fixing couple of odd cases where we are generating odd scroll corrections. 
I just noticed that reaction list in webapp triggers scrollCorrect as redux selector returns a new object if it is an empty list. 

This logic helps not to lock Javascript thread in cases around like this.